src/Telemetry/AzureDevOpsClient/AzureDevOpsClient.cs

```diff
@@ -155,6 +162,60 @@ public async Task<WorkItem> CreateRcaWorkItem(string project, string title, Canc
             return JsonConvert.DeserializeObject<WorkItem>(json);
         }
 
+        public delegate bool TryParseLine(string line, out string parsedValue);
+
+        public async Task<HttpResponseMessage> TryGetLogContents(
+            string logUri,
+            CancellationToken cancellationToken)
+        {
+            for (int i = 0; i < _retryNumber; i++)
+            {
+                using (var request = new HttpRequestMessage(HttpMethod.Get, logUri))
+                {
+                    request.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("text/plain"));
+
+                    try
+                    {
+                        var response = await _httpClient.SendAsync(request, cancellationToken);
```

Comment:
I'd add [HttpCompletionOption.ResponseHeadersRead](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpcompletionoption?view=net-6.0), especially since we expect early aborts, that will let us start reading the stream before the entire content has been buffered, which can be a huge time/memory saver.

Reply:
added